### PR TITLE
Feat/ajax helper

### PR DESCRIPTION
**Pull request by tmkuba**
https://trello.com/c/hoslknmw

### CODE REVIEW

File: client/src/lib/helper.test.js
Comment (leungnamson1023) [inline client/src/lib/helper.test.js]: I know your commit says what you were testing for but just nitpicking at the fact that inside the test it's just blah

File: client/src/components/Reservation.jsx
Comment (matgbx) [inline client/src/components/Reservation.jsx:32]: Is your linter ignoring these files? Just asking considering the use of console.log

File: client/src/components/Reservation.jsx
Comment (matgbx) [inline client/src/components/Reservation.jsx:54]: Nice addition with the prop types

File: client/src/lib/helper.test.js
Comment (matgbx) [inline client/src/lib/helper.test.js]: Should blah? guessing you forgot to update this test before committing - suggest refactor before merge

File: client/src/lib/__mocks__/jquery.js
Comment (matgbx) [inline client/src/lib/__mocks__/jquery.js:33]: Not completely sure what this file is for/does. Can you shed some light?

File: client/src/components/Reservation.test.jsx
Comment (leungnamson1023) [inline client/src/components/Reservation.test.jsx]: though this test is just testing for a snapshot, your actual render passes in the id 100 as the prop but here you pass it in 300, is there a reason why, just curious. is it because its just testing for the div?

**Review by matgbx — CHANGES_REQUESTED** [review]
Looks all good for the most part - small refactor needed on a test in helper.test.js

**Review by leungnamson1023 — COMMENTED** [review]
looks good, your call on changing that test name before merging

File: client/src/components/Reservation.jsx
Comment (tmkuba) [reply client/src/components/Reservation.jsx:32]: Yeah without the console.log my linter will give errors about unused state variables.  So the console log is there to use the state variables because console.log is only a warning and not an error.

File: client/src/lib/__mocks__/jquery.js
Comment (tmkuba) [reply client/src/lib/__mocks__/jquery.js:33]: So this is basically replacing the real $.ajax() with my fake $.ajax() for testing purposes.  I do it so that I can check what my helper function passes in to $.ajax without actually making a real AJAX request.  My fake $.ajax() then returns some fake data and I can run additional tests to make sure my client is processing the returned data correctly (not yet in the test file)

File: client/src/lib/helper.test.js
Comment (tmkuba) [reply client/src/lib/helper.test.js]: good catch, will update to correctly describe the test

File: client/src/lib/helper.test.js
Comment (tmkuba) [reply client/src/lib/helper.test.js]: thank you, will do

File: client/src/components/Reservation.test.jsx
Comment (tmkuba) [reply client/src/components/Reservation.test.jsx]: It was just a placeholder, but I have updated to 100 to match what is passed in by my index.jsx on instantiation

**Comment by leungnamson1023** [comment]
looks good Mark

**Pull request merged**
